genesis/options/recording.py

```diff
@@ -1,29 +1,39 @@
-from genesis.sensors.data_handlers import DataHandler
+from typing import TYPE_CHECKING, Callable
 
 from .options import Options
 
+if TYPE_CHECKING:
+    from genesis.sensors.data_handlers import DataHandler
+
 
 class RecordingOptions(Options):
     """
-    Options for recording data from a sensor.
+    Options for recording data from a data source.
 
     Parameters
     ----------
+    data_func: Callable
+        A function with no arguments that returns the data to be recorded.
     handler: DataHandler
-        The handler that will process the recorded data.
+        The data handler that will process the recorded data.
     hz: float, optional
         The frequency at which to sample data, in Hz (samples per second).
-        If None, the sensor will be sampled every step.
-    buffer_size: int
-        The size of the data queue buffer. Defaults to 0, which means infinite size.
-    buffer_full_wait_time: float
-        The time to wait for buffer space to become available when the buffer is full. Defaults to 0.1 seconds.
+        If None, the data will be sampled every step.
+    run_in_thread: bool, optional
+        Whether to process data in a separate thread. Defaults to True.
```

Comment:
As previously mentioned, the default should be None, and specified by some derived recorder options.